app/src/main/java/com/portfolio/tasky/networking/RequestInterceptor.kt

```diff
@@ -1,19 +1,18 @@
 package com.portfolio.tasky.networking
 
-import com.portfolio.tasky.BuildConfig
+import com.portfolio.tasky.entry.usecases.domain.UserPreferences
 import okhttp3.Interceptor
 import okhttp3.Response
 import java.io.IOException
 import javax.inject.Inject
 
 
-class RequestInterceptor @Inject constructor() :
-    Interceptor {
+class RequestInterceptor @Inject constructor(private val userPreferences: UserPreferences) : Interceptor {
     @Throws(IOException::class)
     override fun intercept(chain: Interceptor.Chain): Response {
         val original = chain.request()
         val requestBuilder = original.newBuilder()
-            .header("x-api-key", BuildConfig.api_key)
+            .header("Authorization", userPreferences.getAuthenticatedUser()?.token as String)
```

Comment:
If the token doesn't exist, your app will crash here, instead I'd add a null check and only add the header if the token exists

Reply:
I've handled this in UserPreference, where it will send an empty, non-null string to avoid that crash. It was getting difficult to check null and deal with non-null variables.

Reply:
But `userPreferences.getAuthenticatedUser()` can apparently return `null` since you have a `?`-check there. So it seems like your prefs are sending a null `AuthenticatedUser` back if there is none (which is correct behavior)